src/lib/command_line.ml

```diff
@@ -80,10 +81,25 @@ module Server = struct
     in
     Log.log t.log ~section:["server"; subsection] json
 
-  let save_job_list t =
-    Storage.Json.save_jsonable t.storage
-      ~path:["server"; "jobs.json"]
-      (`List (List.map t.jobs ~f:(fun j -> `String (Job.id j))))
+  let change_job_list t action =
+    let mutex =
+      match t.job_list_mutex with
+      | Some s -> s
+      | None ->
+        let m = Lwt_mutex.create () in
```

Comment:
> The mutex is there to make sure updates t.jobs <- ... happen together with the Storage.save_jsonable.

If there weren't a mutex, how would this become unsequenced? Isn't the server loop sequential? (I believe you, just trying to figure out where my mental map is going wrong).